### DISCUSSION TITLE
How can I draw a text around a circle in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13778] [Q&A] [answered]
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/d09d0b8c-9b46-41fb-93c9-23fe0fee7b89)

Just like this above.

**Comment by thevortexcloud** [comment, marked as answer]
If you don't need the text to be selectable, you can fairly easily do this with Skia and then save it to a bitmap to display. EG

https://medium.com/medialesson/generating-a-curved-text-in-net-d8a38209c5e2.

**Reply by kyuranger** [reply]
Thank you so much.

**Comment by timunie** [comment]
If that is more or less a static icon, also making this nice looking using Inkscape and save as XAML may be an option.

**Reply by kyuranger** [reply]
The text is mutable so I can't do it your way.

**Comment by Gillibald** [comment]
Most of this is supported by Avalonia: https://learn.microsoft.com/en-us/archive/msdn-magazine/2008/december/foundations-render-text-on-a-path-with-wpf

**Reply by kyuranger** [reply]
Thanks all the same.
